middlewares/profileValidations.js

```diff
@@ -0,0 +1,70 @@
+
+import validation from 'validator';
+
+const Categories = [
+  'Science',
+  'Technology',
+  'Engineering',
+  'Arts',
+  'Mathematic'
+];
+
+export default {
+
+  categoryValidator(req, res, next) {
+    if (Object.keys(req.body).includes('category')) {
+      if (!Categories.includes(req.body.category)) {
+        const allowedList = JSON.stringify(Categories);
+        // eslint-disable-next-line max-len
+        const errorMessage = `Invalid category. Allowed categories are ${allowedList}`;
+        const error = new Error(errorMessage);
+        error.status = 400;
+        return next(error);
+      }
+      return next();
+    }
+    return next();
+  },
+
+  nameValidator(req, res, next) {
+    if (Object.keys(req.body).includes('firstName')) {
+      if (!validation.isAlpha(req.body.firstName)) {
+        const error = new Error('first Name  must be alphabet');
+        error.status = 400;
+        return next(error);
+      }
+    }
+    if (Object.keys(req.body).includes('lastName')) {
+      if (!validation.isAlpha(req.body.lastName)) {
+        const error = new Error('last Name  must be alphabet');
+        error.status = 400;
+        return next(error);
+      }
+    }
+
+    if (Object.keys(req.body).includes('firstName')
```

Comment:
Here, to make this cleaner, let's check for both `firstName` and `lastname` together and push the error message into an array if one or both of them have errors. You can leverage on what we are doing in `requiredParamsValidator` in the `uservalidations`

Reply:
The error message for either of them could be derived from:

`All characters of ${param} must be alphabet`

Reply:
ok